src/services/AccessTokenService.ts

```diff
@@ -99,7 +99,7 @@ class AccessTokenService {
         });
     } else if (config.apiKey === apiKey) {
       // verify the access token
-      next();
+      return next(new ForbiddenException('ApiKey validation unimplemented'));
```

Comment:
If you want to check the apiKey is set - I don't think this is the right place. That should done on startup?

Reply:
2 reasons:
1 - on a default setup with config.apiKey as undefined (or empty string i think), this test passes as long as you pass in some empty api key in the auth header - it just opens up the whole backend if the key is undefined.
2 - if the user intends to use the apikey functionality, then this just says they should implement any logic here (or just call next).
It's a bit safer to throw forbidden by default instead of allowing an undefined apikey (typically the default when not using them) to get through the api.

Reply:
actually not 100% on this, but I can search for the reason this was causing issues - somewhere in our projects you could essentially just pass in some dummy value and get in for free

Reply:
(it could also be a mess up by a devop too to accidentally set a var as an empty string)

Reply:
bump

Reply:
ah yeah, i looked at this issue the other day and forgot to comment... i think its safe to just close it for now - if it comes up again then i'll either re-open or just slap a check in like you suggested. might have just been empty string but havn't really looked into it